libwallet/src/slate.rs

```diff
@@ -250,8 +250,21 @@ impl Slate {
 		tx
 	}
 
-	/// Create a new slate
+	/// Create a new slate with a plain kernel
 	pub fn blank(num_participants: u8, is_invoice: bool) -> Slate {
+		Slate::blank_with_kernel_features(num_participants, is_invoice, 0, None)
+	}
+
+	/// Create a new slate with custom kernel_feature and lock height
+	/// If a lock_height is set kernel_features will have to be either HeightLocked (2) or NoRecentDuplicate (3)
+	/// otherwise the value passed as lock_height will be ignored and a plain kernel will be created
+	/// Invalid arguments for kernel_features (>3) are not allowed and will make the function panic
+	pub fn blank_with_kernel_features(
+		num_participants: u8,
+		is_invoice: bool,
+		kernel_feat_param: u8,
+		lock_height_param: Option<u64>,
+	) -> Slate {
```

Comment:
Would that makes sense to pass a kernel_features like that?
```suggestion
	pub fn blank_with_kernel_features(
		num_participants: u8,
		is_invoice: bool,
		kernel_features: KernelFeatures,
	) -> Slate {
```

That would avoid to have the whole parsing and possible panic in here.

Reply:
I agree with this suggestion, doing it in this way also doesn't need code modification if/when we add new kernel features :+1:

Reply:
@jafalter pinging just in case it was forgotten

Reply:
Thanks, I will do that. But I just need to doublecheck because I remember one of the types was not publicly visible I think that is why I did it like that.

Reply:
@quentinlesceller @phyro the Slate struct requires a `Option<KernelFeaturesArgs>` see:
![image](https://user-images.githubusercontent.com/7274756/115955194-6ac26680-a4f5-11eb-897d-65c9127e3056.png)
`KernelFeaturesArgs` is not a publicly visible struct, therefore can't be passed as the parameter. Using `KernelFeatures` as you recommended would require bigger refactoring of the Slate struct.

Reply:
@jafalter sorry for the late response. I see, in this case I'm ok if we go this way and we can possibly refactor this later :+1:
cc: @quentinlesceller